{
  "timestamp": "2024-02-06 14:14:38.473718",
  "type": "python",
  "version": "1.0",
  "name": "Python Test Suite",
  "description": "Checks subtests and graphics",
  "status": "COMPLETED",
  "result": "FAILED",
  "statusMessage": null,
  "resultMessage": "Some or all Tests failed",
  "details": null,
  "duration": 2.8288395404815674,
  "executionDuration": 2.018667697906494,
  "environment": {
    "Python": "3.10.11",
    "Platform": "Windows-10-10.0.19045-SP0",
    "Packages": {
      "pytest": "7.4.3",
      "pluggy": "1.3.0"
    },
    "Plugins": {
      "metadata": "3.0.0"
    }
  },
  "properties": {
    "test": "i:\\PYTHON\\catester\\examples\\ex2\\test.yaml",
    "specification": "i:\\PYTHON\\catester\\examples\\specification.yaml"
  },
  "debug": {
    "executionDurationStudent": 2.018667697906494,
    "executionDurationReference": 0.0
  },
  "exitcode": "ExitCode.TESTS_FAILED",
  "summary": {
    "total": 4,
    "success": 0,
    "failed": 2,
    "skipped": 2
  },
  "tests": [
    {
      "type": "variable",
      "name": "Test timeout",
      "description": null,
      "setup": [
        "a = 1",
        "b = 1"
      ],
      "teardown": [
        "del a"
      ],
      "status": "TIMEDOUT",
      "result": "FAILED",
      "statusMessage": null,
      "resultMessage": null,
      "details": null,
      "debug": {
        "executionDurationStudent": 2.018667697906494,
        "executionDurationReference": 0,
        "traceback": {}
      },
      "duration": 2.018667697906494,
      "executionDuration": 2.018667697906494,
      "summary": {
        "total": 1,
        "success": 0,
        "failed": 1,
        "skipped": 0
      },
      "tests": [
        {
          "name": "var1",
          "status": "TIMEDOUT",
          "result": "FAILED",
          "resultMessage": null,
          "details": null,
          "debug": {
            "longrepr": "E   Failed: Maximum execution time of 0.3 seconds exceeded",
            "timestamp": 1707225281.5228913
          }
        }
      ]
    },
    {
      "type": "variable",
      "name": "Test variables 1",
      "description": null,
      "setup": null,
      "teardown": null,
      "status": "CRASHED",
      "result": "SKIPPED",
      "statusMessage": null,
      "resultMessage": "Tests skipped",
      "details": null,
      "debug": {
        "executionDurationStudent": 0,
        "executionDurationReference": 0,
        "traceback": {}
      },
      "duration": 0,
      "executionDuration": 0,
      "summary": {
        "total": 2,
        "success": 0,
        "failed": 0,
        "skipped": 2
      },
      "tests": [
        {
          "name": "var1",
          "status": "CRASHED",
          "result": "SKIPPED",
          "resultMessage": "Test skipped",
          "details": null,
          "debug": {
            "longrepr": [
              "i:\\PYTHON\\catester\\catester\\tests\\test_class.py",
              248,
              "Skipped: Success-Dependency `[1]` not satisfied"
            ],
            "timestamp": 1707225281.5228913
          }
        },
        {
          "name": "var2",
          "status": "CRASHED",
          "result": "SKIPPED",
          "resultMessage": "Test skipped",
          "details": null,
          "debug": {
            "longrepr": [
              "i:\\PYTHON\\catester\\catester\\tests\\test_class.py",
              248,
              "Skipped: Success-Dependency `[1]` not satisfied"
            ],
            "timestamp": 1707225281.5228913
          }
        }
      ]
    },
    {
      "type": "variable",
      "name": "Test variables 2",
      "description": null,
      "setup": null,
      "teardown": null,
      "status": "CRASHED",
      "result": "SKIPPED",
      "statusMessage": null,
      "resultMessage": "Tests skipped",
      "details": null,
      "debug": {
        "executionDurationStudent": 0,
        "executionDurationReference": 0,
        "traceback": {}
      },
      "duration": 0,
      "executionDuration": 0,
      "summary": {
        "total": 1,
        "success": 0,
        "failed": 0,
        "skipped": 1
      },
      "tests": [
        {
          "name": "var3",
          "status": "CRASHED",
          "result": "SKIPPED",
          "resultMessage": "Test skipped",
          "details": null,
          "debug": {
            "longrepr": [
              "i:\\PYTHON\\catester\\catester\\tests\\test_class.py",
              248,
              "Skipped: Success-Dependency `['xxx', '1']` not satisfied"
            ],
            "timestamp": 1707225281.5385096
          }
        }
      ]
    },
    {
      "type": "variable",
      "name": "Test crash",
      "description": null,
      "setup": null,
      "teardown": null,
      "status": "CRASHED",
      "result": "FAILED",
      "statusMessage": null,
      "resultMessage": null,
      "details": null,
      "debug": {
        "executionDurationStudent": 0,
        "executionDurationReference": 0,
        "traceback": {
          "name": "<module>",
          "filename": "i:\\PYTHON\\catester\\examples\\ex2\\student\\crash.py",
          "lineno": 4,
          "line": "vahjfg.r1=7",
          "locals": null,
          "errormsg": "name 'vahjfg' is not defined"
        }
      },
      "duration": 0,
      "executionDuration": 0,
      "summary": {
        "total": 1,
        "success": 0,
        "failed": 1,
        "skipped": 0
      },
      "tests": [
        {
          "name": "var1",
          "status": "CRASHED",
          "result": "FAILED",
          "resultMessage": null,
          "details": null,
          "debug": {
            "longrepr": "E   Failed: Execution of i:\\PYTHON\\catester\\examples\\ex2\\student\\crash.py failed",
            "timestamp": 1707225281.5385096
          }
        }
      ]
    }
  ]
}